### DISCUSSION TITLE
Clarification about wildcard to allow domains in team setting and OIDC scope for GitLab

### DISCUSSION THREAD

**Discussion by almereyda** [outline/outline#6438] [Self Hosting] [answered]
It now occurs to me, that I hadn't completely understood how `ALLOWED_DOMAINS` was moved to `/settings/security`. While we tried to use `ALLOWED_DOMAINS=*` before #2678, we're now aware that #3412 / #3489 moved this into the team settings, where we find a happy list of allowed domains.

For clarification, I would like to ask:

Do I interpret the `||` in the `isDomainAllowed` `return` correctly, that all domains will be allowed, when none is configured there?

https://github.com/outline/outline/blob/5d810ac02dbc37a8e9c3a0df2132d8ec33b2c4d8/server/models/Team.ts#L319-L329

If yes, it would be very good to make this more expressive and

- [ ] leave a sentence in https://docs.getoutline.com/s/guide/doc/authentication-Wr4sfjvmL1#h-allowed-domains
- [ ] mention this in the description text  
  ![grafik](https://github.com/outline/outline/assets/1645308/2e0576cb-bdc0-483d-8a96-fc77675c0b58)

**Comment by tommoor** [comment]
> Do I interpret the || in the isDomainAllowed return correctly, that all domains will be allowed, when none is configured there?

That's correct, but it's important to remember this is just an addition level of security to your authentication provider which would be the primary place to restrict access.

**Reply by almereyda** [reply]
Pleased to hear. I'm waiting for user feedback at the moment. My own tests didn't cease the expected results, yet, which means something else might still be at play.

Interestingly, the [*require invites* section](https://docs.getoutline.com/s/guide/doc/authentication-Wr4sfjvmL1#h-require-invites) mentioned in the documentation does not show up on this security settings page.

I'm happy to let anyone in, as long as they passed the registration to the invite-only identity provider in this use case.

**Reply by almereyda** [reply, marked as answer]
Thanks @tommoor for keeping the documentation up to date. As it turns out the `email` scope was missing from our early GitLab OIDC implementation. Was this added at some point?

[GitLab - Outline](https://docs.getoutline.com/s/hosting/doc/gitlab-GjNVvyv7vW) led me in the right direction.

**Reply by tommoor** [reply]
> Interestingly, the [require invites section](https://docs.getoutline.com/s/guide/doc/authentication-Wr4sfjvmL1#h-require-invites) mentioned in the documentation does not show up on this security settings page.

It is not available in the community edition, I added a note to the docs just now
